tests/conftest.py

```diff
@@ -70,3 +73,361 @@ def _sample_all_wsis(_sample_ndpi, _sample_svs, _sample_jp2, tmpdir_factory):
+    """Sample pytest fixture for norm_ruifrok image.
+    Download norm_ruifrok image for pytest."""
+
+    norm_ruifrok_file_path = tmpdir_factory.mktemp("data")\
+        .join("norm_ruifrok.png")
+
+    if not pathlib.Path(norm_ruifrok_file_path).is_file():
+        print("\nDownloading norm_ruifrok image")
+        r = requests.get("https://tiatoolbox.dcs.warwick.ac.uk/testdata/"
+                         "stainnorm/norm_ruifrok.png")
+        with open(norm_ruifrok_file_path, "wb") as f:
+            f.write(r.content)
+    else:
+        print("\nSkipping norm_ruifrok image")
+
+    return norm_ruifrok_file_path
+
+
+@pytest.fixture(scope="session")
+def _norm_vahadane(tmpdir_factory):
+    """Sample pytest fixture for norm_vahadane image.
+    Download norm_vahadane image for pytest."""
+
+    norm_vahadane_file_path = tmpdir_factory.mktemp("data")\
+        .join("norm_vahadane.png")
+
+    if not pathlib.Path(norm_vahadane_file_path).is_file():
+        print("\nDownloading norm_vahadane image")
+        r = requests.get("https://tiatoolbox.dcs.warwick.ac.uk/testdata/"
+                         "stainnorm/norm_vahadane.png")
+        with open(norm_vahadane_file_path, "wb") as f:
+            f.write(r.content)
+    else:
+        print("\nSkipping norm_vahadane image")
+
+    return norm_vahadane_file_path
+
+
+@pytest.fixture(scope="session")
+def _tcga_he_7130(tmpdir_factory):
+    """Sample pytest fixture for TCGA-HE-7130-01Z-00-DX1 image.
+    Download TCGA-HE-7130-01Z-00-DX1 image for pytest."""
+
+    tcga_he_7130_file_path = tmpdir_factory.mktemp("data")\
+        .join("TCGA-HE-7130-01Z-00-DX1.png")
+
+    if not pathlib.Path(tcga_he_7130_file_path).is_file():
+        print("\nDownloading TCGA-HE-7130-01Z-00-DX1 image")
+        r = requests.get("https://tiatoolbox.dcs.warwick.ac.uk/testdata/"
+                         "patchextraction/TCGA-HE-7130-01Z-00-DX1.png")
+        with open(tcga_he_7130_file_path, "wb") as f:
+            f.write(r.content)
+    else:
+        print("\nSkipping TCGA-HE-7130-01Z-00-DX1 image")
+
+    return tcga_he_7130_file_path
+
+
+@pytest.fixture(scope="session")
+def _patch_extr_csv(tmpdir_factory):
+    """Sample pytest fixture for sample patch extraction csv.
+    Download sample patch extraction csv for pytest."""
+
+    csv_file_path = tmpdir_factory.mktemp("data")\
+        .join("sample_patch_extraction.csv")
+
+    if not pathlib.Path(csv_file_path).is_file():
+        print("\nDownloading sample patch extraction csv file")
+        r = requests.get("https://tiatoolbox.dcs.warwick.ac.uk/testdata/"
+                         "patchextraction/sample_patch_extraction.csv")
+        with open(csv_file_path, "wb") as f:
+            f.write(r.content)
+    else:
+        print("\nSkipping sample patch extraction csv file")
+
+    return csv_file_path
+
+
+@pytest.fixture(scope="session")
+def _patch_extr_json(tmpdir_factory):
+    """Sample pytest fixture for sample patch extraction json.
```

Comment:
"""Sample pytest fixture for sample patch extraction json.
    Download sample patch extraction json for pytest.

"""

Reply:
Resolved in this commit https://github.com/TIA-Lab/tiatoolbox/pull/99/commits/f140ef4378c6618ae9d1f5cc7e47785c9c6da702